starlite/plugins/sql_alchemy/config.py

```diff
@@ -0,0 +1,218 @@
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Generic,
+    List,
+    Optional,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from orjson import OPT_SERIALIZE_NUMPY, dumps, loads
+from pydantic import BaseConfig, BaseModel
+from pydantic.generics import GenericModel
+from sqlalchemy.orm import Query, Session
+from typing_extensions import Literal
+
+from starlite.config.logging import BaseLoggingConfig, LoggingConfig
+from starlite.exceptions import MissingDependencyException
+from starlite.middleware.base import DefineMiddleware
+from starlite.utils import default_serializer
+
+try:
+    from sqlalchemy import create_engine
+    from sqlalchemy.engine import Engine
+    from sqlalchemy.ext.asyncio import AsyncEngine, AsyncSession, create_async_engine
+    from sqlalchemy.future import Engine as FutureEngine
+    from sqlalchemy.orm import sessionmaker
+    from sqlalchemy.pool import Pool
+
+    from .middleware import SQLAlchemySessionMiddleware
+except ImportError as e:
+    raise MissingDependencyException("sqlalchemy is not installed") from e
+
+if TYPE_CHECKING:
+    from starlite.datastructures import State
+    from starlite.types import Scope
+
+T = TypeVar("T", Session, AsyncSession)
+
+IsolationLevel = Literal["AUTOCOMMIT", "READ COMMITTED", "READ UNCOMMITTED", "REPEATABLE READ", "SERIALIZABLE"]
+
+
+def serializer(value: Any) -> str:
+    """Serializer for JSON field values.
+
+    Args:
+        value: Any json serializable value.
+
+    Returns:
+        JSON string.
+    """
+    return dumps(
+        value,
+        default=default_serializer,
+        option=OPT_SERIALIZE_NUMPY,
+    ).decode("utf-8")
+
+
+class SQLAlchemySessionConfig(BaseModel):
+    class Config(BaseConfig):
+        arbitrary_types_allowed = True
+
+    bind: Optional[Any] = None
+    autoflush: Optional[bool] = None
+    future: Optional[bool] = None
+    autocommit: Optional[bool] = None
+    expire_on_commit: bool = False
+    twophase: Optional[bool] = None
+    binds: Optional[Any] = None
+    enable_baked_queries: Optional[bool] = None
+    info: Optional[Dict[str, Any]] = None
+    query_cls: Optional[Type[Query]] = None
+
+
+class SQLAlchemyEngineConfig(BaseModel):
+    class Config(BaseConfig):
+        arbitrary_types_allowed = True
+
+    connect_args: Optional[Dict[str, Any]] = None
+    echo: Optional[bool] = None
+    echo_pool: Optional[bool] = None
+    enable_from_linting: Optional[bool] = None
+    future: bool = True
+    hide_parameters: Optional[bool] = None
+    isolation_level: Optional[IsolationLevel] = None
+    json_deserializer: Callable[[str], Any] = loads
```

Comment:
`json_deserializer` is not implemented on all engines.  If the engine doesn't have a native JSON type in SQL, then this parameter will fail.  It should be excluded for those engines.  The `json_serializer` function should exist, so it is likely fine.

Reply:
i added an `engine_supports_json` boolean. The reason is that sqlite does support it, from certain versions, as do other DBs, so this puts it at the hand of the user.